docs/API.md

```diff
@@ -0,0 +1,96 @@
+# API
+
+Before going deeply in the Tartiflette API, don't forget to take a look of the [getting started](./getting-started.md) section.
+
+## Engine initialization
+
+The way to generate the engine is pretty simple.
+
+The engine accepts at least one parameter, called "sdl".
+
+```python
+tartiflette.Engine(sdl)
+```
+
+**When the `sdl` parameters contains the raw schema.**
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    """
+    type Query {
+        hello: String
+    }
+    """
+)
+```
+
+**When the `sdl` parameters target a file.**
+
+The filepath specified have to contain the full schema definition language.
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    "/User/chuck/workspace/mytartiflette/schema.sdl"
+)
+```
+
+**When the `sdl` parameters target a file list.**
+
+Every files will be concatenated, one after the other.
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    [
+        "/User/chuck/workspace/mytartiflette/schema_query.sdl",
+        "/User/chuck/workspace/mytartiflette/schema_mutation.sdl"
+    ]
+)
+```
+
+**When the `sdl` parameters target a folder.**
+
+Every files which are suffixed by `.sdl` will be concatenated, one after the other.
+
+```python
+import tartiflette
+
+engine = tartiflette.Engine(
+    "/User/chuck/workspace/mytartiflette"
+)
+```
+
+### Advanced constructor
+
+The engine provided an advanced interface to be initialized. It accepts optionals and named parameters.
```

Comment:
provides (S not D)
an advanced interface FOR initialization.
It accepts optional and named parameters (no S at optional)